### PR TITLE
Climbing

### PR DESCRIPTION
**Pull request by ShayKalisher1234**
climbing

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/ClimbingSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+   if (instance == null)
+    {
+      instance = new ClimbingSubsystem();
+    }
+    return instance;
+  }
+  
+
+  public ClimbingSubsystem() 
+  {
```
Comment (ori-coval) [inline src/main/java/frc/robot/subsystems/ClimbingSubsystem.java:33]: motor = new motor

File: src/main/java/frc/robot/commands/RightMotorClimbingSetSpeed.java
Comment (ori-coval) [inline src/main/java/frc/robot/commands/RightMotorClimbingSetSpeed.java:1]: use instant command

File: src/main/java/frc/robot/commands/LeftMotorClimbingSetSpeed.java
Comment (ori-coval) [inline src/main/java/frc/robot/commands/LeftMotorClimbingSetSpeed.java:1]: use instant command

File: src/main/java/frc/robot/commands/setSpeedTrap.java
Comment (ori-coval) [inline src/main/java/frc/robot/commands/setSpeedTrap.java:1]: use instant command

File: src/main/java/frc/robot/subsystems/SubsystemTrap.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+   if (instance == null)
+    {
+      instance = new SubsystemTrap();
+    }
+    return instance;
+  }
+
+
+  /** Creates a new SubsystemTrap. */
+  private SubsystemTrap() {
```
Comment (ori-coval) [inline src/main/java/frc/robot/subsystems/SubsystemTrap.java:31]: motor = new motor

File: src/main/java/frc/robot/subsystems/SubsystemTrap.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  }
+  public void setSpeed(double speed){
+    trapMotor = new WPI_TalonSRX(DEVICE_NUMBER);
```
Comment (ori-coval) [inline src/main/java/frc/robot/subsystems/SubsystemTrap.java]: `trapMotor = new WPI_TalonSRX(DEVICE_NUMBER);` should be in the constractur

File: src/main/java/frc/robot/commands/ClimbingMotorsSetSpeed.java
Comment (ori-coval) [inline src/main/java/frc/robot/commands/ClimbingMotorsSetSpeed.java:1]: use instant command

File: src/main/java/frc/robot/subsystems/SubsystemTrap.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import com.ctre.phoenix.motorcontrol.TalonSRXControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.ctre.phoenix6.hardware.TalonFX;
+import static frc.robot.Constants.trapConstants.*;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class SubsystemTrap extends SubsystemBase {
+  private WPI_TalonSRX trapMotor;
```
Comment (ori-coval) [inline src/main/java/frc/robot/subsystems/SubsystemTrap.java]: change the name to m_trapMotor

File: src/main/java/frc/robot/Constants.java [hunk 5]
```diff
@@ -223,4 +223,9 @@ public static class Vision {
 
     public static final Pose2d target = new Pose2d(1, 1, new Rotation2d(Units.degreesToRadians(0)));
   }
+  public static class trapConstants {
+    public static final int DEVICE_NUMBER = 0;
```
Comment (ori-coval) [inline src/main/java/frc/robot/Constants.java]: change to trapMotorID

**Review by ori-coval — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/ClimbingSubsystem.java (hunk 1 above)
Comment (ShayKalisher1234) [reply src/main/java/frc/robot/subsystems/ClimbingSubsystem.java:33]: fixed

File: src/main/java/frc/robot/subsystems/SubsystemTrap.java (hunk 2 above)
Comment (ShayKalisher1234) [reply src/main/java/frc/robot/subsystems/SubsystemTrap.java:31]: fixed

File: src/main/java/frc/robot/subsystems/SubsystemTrap.java (hunk 3 above)
Comment (ShayKalisher1234) [reply src/main/java/frc/robot/subsystems/SubsystemTrap.java]: fixed

File: src/main/java/frc/robot/subsystems/SubsystemTrap.java (hunk 4 above)
Comment (ShayKalisher1234) [reply src/main/java/frc/robot/subsystems/SubsystemTrap.java]: fixed

File: src/main/java/frc/robot/Constants.java (hunk 5 above)
Comment (ShayKalisher1234) [reply src/main/java/frc/robot/Constants.java]: fixed

File: src/main/java/frc/robot/commands/LeftMotorClimbingSetSpeed.java
Comment (ShayKalisher1234) [reply src/main/java/frc/robot/commands/LeftMotorClimbingSetSpeed.java:1]: fixed

File: src/main/java/frc/robot/commands/RightMotorClimbingSetSpeed.java
Comment (ShayKalisher1234) [reply src/main/java/frc/robot/commands/RightMotorClimbingSetSpeed.java:1]: fixed

File: src/main/java/frc/robot/commands/setSpeedTrap.java
Comment (ShayKalisher1234) [reply src/main/java/frc/robot/commands/setSpeedTrap.java:1]: fixed

File: src/main/java/frc/robot/commands/ClimbingMotorsSetSpeed.java
Comment (ShayKalisher1234) [reply src/main/java/frc/robot/commands/ClimbingMotorsSetSpeed.java:1]: fixed

**Pull request closed without merging**